src/explore-education-statistics-common/src/modules/table-tool/utils/__tests__/createExpandedRowHeaders.test.ts

```diff
@@ -0,0 +1,1549 @@
+import Header from '../../components/utils/Header';
+import createExpandedRowHeaders from '../createExpandedRowHeaders';
+
+describe('createExpandedRowHeaders', () => {
+  test('should return an empty array if no row headers are provided', () => {
+    expect(createExpandedRowHeaders([])).toEqual([]);
+  });
+
+  test('should return a single row of row headers if no groups are provided', () => {
+    const rowHeaders: Header[] = [
+      new Header('1', '1'),
+      new Header('2', '2'),
+      new Header('3', '3'),
+    ];
+
+    expect(createExpandedRowHeaders(rowHeaders)).toEqual([
+      [
+        {
+          crossSpan: 1,
+          id: '1',
+          isGroup: false,
+          span: 1,
+          text: '1',
+        },
+      ],
+      [
+        {
+          crossSpan: 1,
+          id: '2',
+          isGroup: false,
+          span: 1,
+          text: '2',
+        },
+      ],
+      [
+        {
+          crossSpan: 1,
+          id: '3',
+          isGroup: false,
+          span: 1,
+          text: '3',
+        },
+      ],
+    ]);
+  });
+
+  test('should return multiple rows of row headers if groups are provided', () => {
+    const rowHeaders: Header[] = [
+      new Header('1', '1'),
+      new Header('2', '2'),
+      new Header('3', '3').addChild(new Header('2.1', '2.1')),
+      new Header('4', '4').addChild(new Header('3.1', '3.1')),
+    ];
+
+    expect(createExpandedRowHeaders(rowHeaders)).toEqual([
+      [
+        {
+          crossSpan: 1,
+          id: '1',
+          isGroup: false,
+          span: 1,
+          text: '1',
+        },
+      ],
+      [
+        {
+          crossSpan: 1,
+          id: '2',
+          isGroup: false,
+          span: 1,
+          text: '2',
+        },
+      ],
+      [
+        {
+          crossSpan: 1,
+          id: '3',
+          isGroup: true,
+          span: 1,
+          text: '3',
+        },
+        {
+          crossSpan: 1,
+          id: '2.1',
+          isGroup: false,
+          span: 1,
+          text: '2.1',
+        },
+      ],
+      [
+        {
+          crossSpan: 1,
+          id: '4',
+          isGroup: true,
+          span: 1,
+          text: '4',
+        },
+        {
+          crossSpan: 1,
+          id: '3.1',
+          isGroup: false,
+          span: 1,
+          text: '3.1',
+        },
+      ],
+    ]);
+  });
+
+  describe('returns correct headers with multiple levels of headers', () => {
+    test('renders 2x2 table correctly', () => {
+      const rowHeaders: Header[] = [
+        new Header('1', '1')
+          .addChild(new Header('3', '3'))
+          .addChild(new Header('4', '4')),
+        new Header('2', '2')
+          .addChild(new Header('3', '3'))
+          .addChild(new Header('4', '4')),
+      ];
+      const expandedRowHeaders = [
```

Comment:
Would add an explicit type here i.e. `ExpandedHeader[][]` so we get feedback about type problems at compile time instead of runtime.

Same applies for the rest of these test cases.